docs/prd/command-registry-pattern.md

```
@@ -1277,6 +1277,223 @@ func TestDeeplyNestedCommandExecution(t *testing.T) {
+🚩 **These patterns indicate a function should move to `cmd/`:**
+
+- Function name matches command name: `ExecuteVersionList` → probably cmd/version-specific
+- Function only called from one command package
+- Function has 5+ parameters that are all CLI flags
+- Function exists just to call other functions in the same package
+- Test file only has one test that calls the command
+
+### Benefits of Self-Contained Command Packages
+
+**Before (scattered logic):**
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)